jsettlers.logic/src/main/java/jsettlers/logic/movable/strategies/specialists/PioneerStrategy.java

```diff
@@ -87,22 +89,22 @@ private void findWorkablePosition() {
 		}
 	}
 
-	private final EDirection getCloseForeignTile() {
-		EDirection bestNeighbourDir = null;
-		double bestNeighbourDistance = Double.MAX_VALUE; // distance from start point
+	private EDirection getCloseForeignTile() {
+		EDirection[] bestNeighbourDir = new EDirection[1];
```

Comment:
You can use a custom Class here.

You can add the visiting method to that object to make the code clean and save some object creations.